src/Microsoft.Diagnostics.Monitoring.WebApi/Controllers/DiagController.cs

```diff
@@ -672,21 +682,40 @@ private static LogFormat ComputeLogFormat(IList<MediaTypeHeaderValue> acceptedHe
 
             if (string.IsNullOrEmpty(providerName))
             {
-                return new OutputStreamResult(
+                return Task.FromResult<ActionResult>(new OutputStreamResult(
                     action,
                     contentType,
                     asAttachment ? fileName : null,
-                    scope);
+                    scope));
             }
             else
             {
-                return new EgressStreamResult(
+                return SendToEgress(new EgressOperation(
                     action,
                     providerName,
                     fileName,
                     endpointInfo,
                     contentType,
-                    scope);
+                    scope),
+                    limitKey: artifactType);
+            }
+        }
+
+        private async Task<ActionResult> SendToEgress(EgressOperation egressStreamResult, string limitKey)
+        {
+            try
+            {
+                Guid operationId = await _operationsStore.AddOperation(egressStreamResult, limitKey);
+                string newUrl = this.Url.Action(
+                    action: nameof(OperationController.GetOperationStatus),
+                    controller: OperationController.ControllerName, new { operationId = operationId },
+                    protocol: this.HttpContext.Request.Scheme, this.HttpContext.Request.Host.ToString());
+
+                return Accepted(newUrl);
+            }
+            catch (TooManyRequestsException)
+            {
+                return this.StatusCode(StatusCodes.Status429TooManyRequests);
```

Comment:
Note this does not work as expected for netcore 3.1 (it works fine for 5.0). It appears that StatusCodeResult gets converted to ObjectResult, and then there's a content type mismatch. This produces a 406.

Reply:
Thinking about this some more, I think we are going up against the declared api controller behavior, which is to produce a problemdetails. My tentative thinking is to get rid of the middleware altogether and let the exception throw and then get converted to an appropriate problemdetails.